src/smt/set_defaults.cpp

```diff
@@ -92,12 +92,14 @@ void setDefaults(LogicInfo& logic, bool isInternalSubsolver)
     }
   }
   if (options::unsatCoresNew()
+      && options::unsatCoresMode() != options::UnsatCoresMode::FULL_PROOF
```

Comment:
This block of code should check `options::unsatCoresNew() && options::produceProofs()`
and set `unsatCoresMode` to FULL_PROOF if so, with a `Notice` if the unsatCoreMode was set by user.

It should be run *after* the next block of code (and any other block of code that could set `produceProofs` to true).

Reply:
Good point! I changed accordingly.